inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditReporterImpl.java

```diff
@@ -54,19 +58,36 @@ public class AuditReporterImpl implements Serializable {
     private static final int BATCH_NUM = 100;
     private final ReentrantLock GLOBAL_LOCK = new ReentrantLock();
     private static final int PERIOD = 1000 * 60;
-    private final ConcurrentHashMap<String, StatInfo> countMap = new ConcurrentHashMap<>();
-    private final ConcurrentHashMap<String, StatInfo> threadCountMap = new ConcurrentHashMap<>();
-    private final ConcurrentHashMap<String, StatInfo> deleteCountMap = new ConcurrentHashMap<>();
-    private final List<String> deleteKeyList = new ArrayList<>();
+    private final ConcurrentHashMap<Long, ConcurrentHashMap<String, StatInfo>> preStatMap =
+            new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<Long, ConcurrentHashMap<String, StatInfo>> summaryStatMap =
+            new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<Long, ConcurrentHashMap<String, StatInfo>> expiredStatMap =
+            new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<Long, List<String>> expiredKeyList = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<Long, Long> flushTime = new ConcurrentHashMap<>();
     private final Config config = new Config();
     private int packageId = 1;
     private int dataId = 0;
     private boolean initialized = false;
     private SenderManager manager;
+    private AtomicInteger flushStat = new AtomicInteger(0);
 
     private final ScheduledExecutorService timeoutExecutor = Executors.newSingleThreadScheduledExecutor();
     private AuditConfig auditConfig = null;
     private SocketAddressListLoader loader = null;
+    private static final long DEFAULT_ISOLATE_KEY = 0;
+    private int flushStatThreshold = 100;
+
+    public void setFlushStatThreshold(int flushStatThreshold) {
+        this.flushStatThreshold = flushStatThreshold;
+    }
+
+    public void setAutoFlush(boolean autoFlush) {
+        this.autoFlush = autoFlush;
+    }
+
+    private boolean autoFlush = true;
```

Comment:
Member variables need to be declared before all member methods.

Reply:
resolved